apps/internal/oauth/oauth.go

```diff
@@ -194,14 +195,31 @@ func (d DeviceCode) Token(ctx context.Context) (accesstokens.TokenResponse, erro
 	}
 }
 
-var waitRE = regexp.MustCompile("(authorization_pending|slow_down)")
+type deviceCodeError struct {
```

Comment:
Don't we receive a standard, structured error response for all failed requests?  Assuming that's the case, we should have a standard error type so we don't have to create these one-off types.  In `azidentity` we called this [`AADAuthenticationFailedError `](https://github.com/Azure/azure-sdk-for-go/blob/master/sdk/azidentity/azidentity.go#L45).  We can leave that for a separate PR though.

Reply:
I had this conversation with @element-of-surprise while doing the error design doc, of having a standard error type for the server response, but since in most cases the error message can sufficiently tell what needs to change, we decided to not go for it and instead have a `CallErr` so that in the case they need to use something they can get the response body.

Reply:
If this is the only time we need to unmarshall the error info to check some value then we can leave it as-is.  I just want to ensure we're not adding types that essentially do the same thing because there's no shared definition.

Reply:
For now, this is the only place we need to unmarshall to check some value. There might be some features where we need to unmarshall the error to provide specific error info to the app developer. For that this is what we plan to do, quoting from the error design doc
``` If further differentiation is required, we can add custom errors that use Go error wrapping on top of CallErr to achieve our diagnostic goals (such as detecting when to retry a call due to transient errors).``` 

I guess, we can add a type even internally if such a thing is needed in more cases.